### PR TITLE
Created responsive product grid using test data

### LINKED ISSUE #12: feature/create-item-grid
## Brief Description
Creating front-end item grid for the customer to browse items.

## Feature Requirements
-  Renders an item component for each item in filtered database
-  Item component contains:
   - Name
   - Image
   - Description
   - Price
   - Item ID
   - Seller
   - Brand
   - Add to Cart button

### PR DESCRIPTION
**Pull request by Adamo-O**
## Brief Description
Created product grid that renders products from test JSON data.

## Linked Issues
- closes #12 

## Changes
- Created ProductGrid.js which encapsulates all Product Grid functions and logic.
   - ProductGrid component renders inputted products using the prop named `products`
   - ProductPreview component is created for each product
      - Uses Card, CardActions, Button, and Grid components from MUI
- `ProductGrid` added to App.js to render Product Grid
- Added TestProducts.json with test data before database implemented. Products contain:
   - ID
   - Name
   - Image (Link)
   - Description
   - Seller
   - Price
- Added ProductGrid unit test to check if product names displayed on screen.

## Additional context
What should be changed in the future:
- Product data
   - Can be changed directly from App.js
- Colours
- Button design
- Button links/actions
   - Add to Cart
   - View More (if needed)
- Grid spacing, sizing, layout (if needed)

### CODE REVIEW

File: online-shopping-website/src/App.css [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+    border-color: rgb(78, 219, 78);
+    transition: 0.3s all ease-in-out;
+  }
+
+  .ProductButtonContained {
+    background-color: rgb(72, 136, 72);
+  }
+
+  .ProductButton:hover, .ProductButtonContained:hover {
+    background-color: rgba(79, 255, 108, 0.198);
```
Comment (aspyridakos) [inline online-shopping-website/src/App.css:38]: Question: How is `rgba` different from `rgb`?

Comment (Adamo-O) [reply online-shopping-website/src/App.css:38]: The _a_ in `rgba` stands for Alpha, which is used for opacity. On top of adjusting the colors using `rgb`, we can adjust opacity as well, to allow for transparency. I did it here to make the hover color more subtle. It's pretty great!

**Review by aspyridakos — APPROVED** [review]
Solid 👍

File: online-shopping-website/src/App.css (hunk 1 above)
Comment (aspyridakos) [reply online-shopping-website/src/App.css:38]: Nice! Sounds like a super useful feature, definitely will be using it in the future :)

File: online-shopping-website/src/Components/ProductGrid.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+// Creates a Grid item with a ProductPreview for each product in inputted data. Sets element keys to product ids.
+const iterateProduct = (data) => {
+  return data.products.map(product => {
+    return (
+      <Grid item key={product.id} xs={12} sm={6} lg={4}>
+        <ProductPreview key={product.id} product={product} />
+      </Grid>
+    );
+  })
+}
```
Comment (aspyridakos) [inline online-shopping-website/src/Components/ProductGrid.js:45]: Could this code be tested (we should add a unit test file as well to include all our function tests)?

Comment (aspyridakos) [reply online-shopping-website/src/Components/ProductGrid.js:45]: Same goes from `renderProductGrid`

**Review by aspyridakos — CHANGES_REQUESTED** [review]

File: online-shopping-website/src/TestProducts.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      "seller": "MacDon't",
+      "price": 8
+    },
+    {
+      "id": 4,
+      "name": "Cupcake",
+      "image": "https://dufflet.com/wp-content/uploads/2021/05/hero_cup_CVANCM_Dec2013.png",
+      "description": "As sweet as I don't know what.",
+      "seller": "Alan Wake's Cupcakes",
+      "price": 12
```
Comment (aspyridakos) [inline online-shopping-website/src/TestProducts.json:41]: Don't forget to add the brand here as well!

**Review by aspyridakos — CHANGES_REQUESTED** [review]

File: online-shopping-website/src/Components/ProductGrid.js (hunk 2 above)
Comment (Adamo-O) [reply online-shopping-website/src/Components/ProductGrid.js:46]: We can do some unit tests, sure! What would the condition be though? Just making sure that something was rendered to the screen? Like making sure the text is visible or something?

Comment (aspyridakos) [reply online-shopping-website/src/Components/ProductGrid.js:46]: Yup exactly. Take a lot at template tests online, it should be similar to what you'd be looking for here.

Comment (Adamo-O) [reply online-shopping-website/src/Components/ProductGrid.js:46]: I replaced `renderProductGrid` with the `ProductGrid` component for easier testing and best practices. I added the unit test to App.test.js, and it checks the screen to ensure all product names are visible. We can add different tests later but this helps us narrow down where a future issue may be coming from.

**Review by emma-amar — APPROVED** [review]
looks good to me! also I mostly wrote tests at my internship so I can definitely take that on when we get there :-)

**Review by aspyridakos — APPROVED** [review]

**Pull request merged**